plugin/trino-geospatial/src/main/java/io/trino/plugin/geospatial/GeoFunctions.java

```diff
@@ -1188,6 +1191,34 @@ public static Double stDistance(@SqlType(GEOMETRY_TYPE_NAME) Slice left, @SqlTyp
         return leftGeometry.isEmpty() || rightGeometry.isEmpty() ? null : leftGeometry.distance(rightGeometry);
     }
 
+    @SqlNullable
+    @Description("Return the closest points on the two geometries")
+    @ScalarFunction("geometry_nearest_points")
+    @SqlType("row(" + GEOMETRY_TYPE_NAME + "," + GEOMETRY_TYPE_NAME + ")")
+    public static Block geometryNearestPoints(@SqlType(GEOMETRY_TYPE_NAME) Slice left, @SqlType(GEOMETRY_TYPE_NAME) Slice right)
+    {
+        RowType rowType = RowType.anonymous(ImmutableList.of(GEOMETRY, GEOMETRY));
+        PageBuilder pageBuilder = new PageBuilder(ImmutableList.of(rowType));
```

Comment:
Move this below the `if (leftGeometry.isEmpty() || rightGeometry.isEmpty()) {` check below to avoid constructing these objects in the case of empty input geometries.

Reply:
Good point. Fixed.